src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs

```diff
@@ -94,5 +113,57 @@ internal class ThreadInfo
             public TimeSpan PrivilegedProcessorTime { get; set; }
             public TimeSpan UserProcessorTime { get; set; }
         }
+
+        public static void OutputResultToStream(StreamWriter sw, HashSet<string> threadIds = null, bool includeStackObjects = false)
+        {
+            var ravenDebugExec = Path.Combine(AppContext.BaseDirectory, "Raven.Debug.exe");
+            if (File.Exists(ravenDebugExec) == false)
+                throw new FileNotFoundException($"Could not find debugger tool at '{ravenDebugExec}'");
+
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                var sb = new StringBuilder($"stack-traces --pid {currentProcess.Id}");
+
+                if (threadIds != null && threadIds.Count > 0)
+                {
+                    foreach (var threadId in threadIds)
+                    {
+                        sb.Append($" --tid {threadId}");
+                    }
+                }
+
+                if (includeStackObjects)
+                    sb.Append(" --includeStackObjects");
+
+                var process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        Arguments = sb.ToString(),
+                        FileName = ravenDebugExec,
+                        WindowStyle = ProcessWindowStyle.Normal,
+                        LoadUserProfile = false,
+                        RedirectStandardError = true,
+                        RedirectStandardOutput = true,
+                        UseShellExecute = false
+                    },
+                    EnableRaisingEvents = true
+                };
+
+                sb.Clear();
+                process.OutputDataReceived += (sender, args) => sw.Write(args.Data);
+                process.ErrorDataReceived += (sender, args) => sb.Append(args.Data);
```

Comment:
`sb.Append` on error? Did you mean `sw.Write`?

Reply:
Do we need to worry about concurrency here?

Reply:
I meant to use the sb to throw the error message. Fixed.